packages/mailer/core/src/mailer.ts

```diff
@@ -0,0 +1,347 @@
+      ? 'development'
+      : 'production'
+
+    // Config
+    this.handlers = this.config.handling.handlers
+    this.renderers = this.config.rendering.renderers
+
+    // Extract defaults from config
+    this.defaults = extractDefaults(this.config.defaults ?? {})
+
+    // Validate handlers for test and development modes
+    const testHandlerKey = this.config.test?.handler
+    if (testHandlerKey === undefined) {
+      // TODO: Attempt to use a default in-memory handler if the required package is installed
+      //       otherwise default to null and log a warning
+      this.config.test = {
+        ...this.config.test,
+        handler: null,
+      }
+      this.logger.warn(
+        'The test handler is null, this will prevent mail from being processed in test mode'
+      )
+    } else if (testHandlerKey === null) {
+      this.logger.warn(
+        'The test handler is null, this will prevent mail from being processed in test mode'
+      )
+    } else {
+      if (this.handlers[testHandlerKey] === undefined) {
+        throw new Error(
+          `The specified test handler '${testHandlerKey.toString()}' is not defined`
+        )
+      }
+    }
+    const developmentHandlerKey = this.config.development?.handler
+    if (developmentHandlerKey === undefined) {
+      // TODO: Attempt to use a default studio handler if the required package is installed
+      //       otherwise default to null and log a warning
+      this.config.development = {
+        ...this.config.development,
+        handler: null,
+      }
+      this.logger.warn(
+        'The development handler is null, this will prevent mail from being processed in development mode'
+      )
+    } else if (developmentHandlerKey === null) {
+      this.logger.warn(
+        'The development handler is null, this will prevent mail from being processed in development mode'
+      )
+    } else {
+      if (this.handlers[developmentHandlerKey] === undefined) {
+        throw new Error(
+          `The specified development handler '${developmentHandlerKey.toString()}' is not defined`
+        )
+      }
+    }
+
+    // Validate default handler and renderer
+    const defaultHandlerKey = this.config.handling.default
+    if (defaultHandlerKey === undefined) {
+      throw new Error('No default handler configured')
+    } else if (this.handlers[defaultHandlerKey] === undefined) {
+      throw new Error(
+        `The specified default handler '${defaultHandlerKey.toString()}' is not defined`
+      )
+    }
+    const defaultRendererKey = this.config.rendering.default
+    if (defaultRendererKey === undefined) {
+      throw new Error('No default renderer configured')
+    } else if (this.renderers[defaultRendererKey] === undefined) {
+      throw new Error(
+        `The specified default renderer '${defaultRendererKey.toString()}' is not defined`
+      )
+    }
+
+    // Initial logging
+    this.logger.debug({}, `Mailer initialized in ${this.mode} mode`)
```

Comment:
As I noted in a chat, perhaps including what default handler and render will be used in dev might be helpful?